plugins/modules/azure_rm_virtualmachine.py

```diff
@@ -86,6 +86,28 @@
             - A valid Azure VM size value. For example, C(Standard_D4).
             - Choices vary depending on the subscription and location. Check your subscription for available choices.
             - Required when creating a VM.
+    priority:
+        description:
+            - Priority of the VM.
+        choices:
+            - Regular
+            - Spot
+        default: Regular
```

Comment:
argument_spec defines default as (None) but documentation defines default as ('Regular')

Reply:
Regular is the equivalent of None. To avoid any confusion, changed to 'None' by default and added a mention of setting None is the equivalent of setting up a Regular VM (non Spot)